File: Models/DSSAT_ST_standalone/src/java/DSSAT_ST_standalone/STEMP_Component.java
public class STEMP_Component
{
    
    public STEMP_Component() { }

    STEMP _STEMP = new STEMP();

    public Double getMSALB()
    { return _STEMP.getMSALB(); }
    public void setMSALB(Double _MSALB){
    _STEMP.setMSALB(_MSALB);
    }

    public Integer getNL()
    { return _STEMP.getNL(); }
    public void setNL(Integer _NL){
    _STEMP.setNL(_NL);
    }

    public Double [] getLL()
    { return _STEMP.getLL(); }
    public void setLL(Double [] _LL){
    _STEMP.setLL(_LL);
    }

    public Integer getNLAYR()
    { return _STEMP.getNLAYR(); }
    public void setNLAYR(Integer _NLAYR){
    _STEMP.setNLAYR(_NLAYR);
    }

    public Double [] getDS()
    { return _STEMP.getDS(); }
    public void setDS(Double [] _DS){
    _STEMP.setDS(_DS);
    }

    public Double [] getDLAYR()
    { return _STEMP.getDLAYR(); }
    public void setDLAYR(Double [] _DLAYR){
    _STEMP.setDLAYR(_DLAYR);
    }

    public String getISWWAT()
    { return _STEMP.getISWWAT(); }
    public void setISWWAT(String _ISWWAT){
    _STEMP.setISWWAT(_ISWWAT);
    }

    public Double [] getBD()
    { return _STEMP.getBD(); }
    public void setBD(Double [] _BD){
    _STEMP.setBD(_BD);
    }

    public Double [] getSW()
    { return _STEMP.getSW(); }
    public void setSW(Double [] _SW){
    _STEMP.setSW(_SW);
    }

    public Double getXLAT()
    { return _STEMP.getXLAT(); }
    public void setXLAT(Double _XLAT){
    _STEMP.setXLAT(_XLAT);
    }

    public Double [] getDUL()
    { return _STEMP.getDUL(); }
    public void setDUL(Double [] _DUL){
    _STEMP.setDUL(_DUL);
    }
    public void  Calculate_Model(STEMP_State s, STEMP_State s1, STEMP_Rate r, STEMP_Auxiliary a, STEMP_Exogenous ex)
    {
        _STEMP.Calculate_Model(s, s1, r, a, ex);
    }
    private Double MSALB;
    private Integer NL;
    private Double [] LL;
    private Integer NLAYR;
    private Double [] DS;
    private Double [] DLAYR;
    private String ISWWAT;
    private Double [] BD;
    private Double [] SW;
    private Double XLAT;
    private Double [] DUL;
    public STEMP_Component(STEMP_Component toCopy) // copy constructor 
    {
        this.MSALB = toCopy.getMSALB();
        this.NL = toCopy.getNL();
        
        for (int i = 0; i < toCopy.getLL().length; i++)
        {
            LL[i] = toCopy.getLL()[i];
        }
        this.NLAYR = toCopy.getNLAYR();
        
        for (int i = 0; i < toCopy.getDS().length; i++)
        {
            DS[i] = toCopy.getDS()[i];
        }
        
        for (int i = 0; i < toCopy.getDLAYR().length; i++)
        {
            DLAYR[i] = toCopy.getDLAYR()[i];
        }
        this.ISWWAT = toCopy.getISWWAT();
        
        for (int i = 0; i < toCopy.getBD().length; i++)
        {
            BD[i] = toCopy.getBD()[i];
        }
        
        for (int i = 0; i < toCopy.getSW().length; i++)
        {
            SW[i] = toCopy.getSW()[i];
        }
        this.XLAT = toCopy.getXLAT();
        
        for (int i = 0; i < toCopy.getDUL().length; i++)
        {
            DUL[i] = toCopy.getDUL()[i];
        }

    }
}

File: Models/DSSAT_ST_standalone/src/cpp/DSSAT_ST_standalone/STEMP_Component.cpp
#include "STEMP_Component.h"
using namespace DSSAT_ST_standalone;
STEMP_Component::STEMP_Component()
{
       
}


double STEMP_Component::getMSALB(){ return this->MSALB; }
int STEMP_Component::getNL(){ return this->NL; }
std::vector<double> & STEMP_Component::getLL(){ return this->LL; }
int STEMP_Component::getNLAYR(){ return this->NLAYR; }
std::vector<double> & STEMP_Component::getDS(){ return this->DS; }
std::vector<double> & STEMP_Component::getDLAYR(){ return this->DLAYR; }
std::string STEMP_Component::getISWWAT(){ return this->ISWWAT; }
std::vector<double> & STEMP_Component::getBD(){ return this->BD; }
std::vector<double> & STEMP_Component::getSW(){ return this->SW; }
double STEMP_Component::getXLAT(){ return this->XLAT; }
std::vector<double> & STEMP_Component::getDUL(){ return this->DUL; }

void STEMP_Component::setMSALB(double _MSALB)
{
    _STEMP.setMSALB(_MSALB);
}
void STEMP_Component::setNL(int _NL)
{
    _STEMP.setNL(_NL);
}
void STEMP_Component::setLL(const std::vector<double> & _LL)
{
    _STEMP.setLL(_LL);
}
void STEMP_Component::setNLAYR(int _NLAYR)
{
    _STEMP.setNLAYR(_NLAYR);
}
void STEMP_Component::setDS(const std::vector<double> & _DS)
{
    _STEMP.setDS(_DS);
}
void STEMP_Component::setDLAYR(const std::vector<double> & _DLAYR)
{
    _STEMP.setDLAYR(_DLAYR);
}
void STEMP_Component::setISWWAT(std::string _ISWWAT)
{
    _STEMP.setISWWAT(_ISWWAT);
}
void STEMP_Component::setBD(const std::vector<double> & _BD)
{
    _STEMP.setBD(_BD);
}
void STEMP_Component::setSW(const std::vector<double> & _SW)
{
    _STEMP.setSW(_SW);
}
void STEMP_Component::setXLAT(double _XLAT)
{
    _STEMP.setXLAT(_XLAT);
}
void STEMP_Component::setDUL(const std::vector<double> & _DUL)
{
    _STEMP.setDUL(_DUL);
}
void STEMP_Component::Calculate_Model(STEMP_State &s, STEMP_State &s1, STEMP_Rate &r, STEMP_Auxiliary &a, STEMP_Exogenous &ex)
{
    _STEMP.Calculate_Model(s, s1, r, a, ex);
}
STEMP_Component::STEMP_Component(STEMP_Component& toCopy)
{
    MSALB = toCopy.getMSALB();
    NL = toCopy.getNL();
    for (int i = 0; i < NL; i++)
{
    LL[i] = toCopy.getLL()[i];
}

    NLAYR = toCopy.getNLAYR();
    for (int i = 0; i < NL; i++)
{
    DS[i] = toCopy.getDS()[i];
}

    for (int i = 0; i < NL; i++)
{
    DLAYR[i] = toCopy.getDLAYR()[i];
}

    ISWWAT = toCopy.getISWWAT();
    for (int i = 0; i < NL; i++)
{
    BD[i] = toCopy.getBD()[i];
}

    for (int i = 0; i < NL; i++)
{
    SW[i] = toCopy.getSW()[i];
}

    XLAT = toCopy.getXLAT();
    for (int i = 0; i < NL; i++)
{
    DUL[i] = toCopy.getDUL()[i];
}

}